File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/IUsedeskActionListener.java
package ru.usedesk.chat_sdk.external.entity;

import androidx.annotation.NonNull;

import java.util.List;

import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskException;

public interface IUsedeskActionListener {

    void onConnected();

    void onMessageReceived(UsedeskMessage message);

    void onMessagesReceived(List<UsedeskMessage> messages);

    void onFeedbackReceived();

    void onOfflineFormExpected();

    void onDisconnected();

    void onException(@NonNull UsedeskException usedeskException);
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/IUsedeskActionListener.kt
package ru.usedesk.chat_sdk.external.entity

interface IUsedeskActionListener {
    fun onConnected()

    fun onChatItemReceived(chatItem: UsedeskChatItem)

    fun onChatItemsReceived(chatItems: List<UsedeskChatItem>)

    fun onFeedbackReceived()

    fun onOfflineFormExpected(chatConfiguration: UsedeskChatConfiguration)

    fun onDisconnected()

    fun onException(usedeskException: Exception)
}